ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto

```diff
@@ -82,21 +92,67 @@ message SubmitAndWaitRequest {
 
 }
 
+message SubmitAndWaitResponse {
+  // The completion offset for the submission that triggered this response.
+  // This offset can be used as the submission rank for subsequent command submissions.
+  // Optional
+  // TODO Make it required for responses with status code `OK` once all Daml ledgers support it.
+  string completion_offset = 1;
+
+  // Whether this response is a definite answer for the deduplication and rank guarantees,
+  // as defined in ``commands.proto``.
+  // Always true if the status code is `OK`.
+  bool definite_answer = 2;
```

Comment:
We want to return these two fields especially when the status code is not OK (so that the application gets hold of a higher offset for the next submission rank / can make a decision to not resubmit). I'm not sure whether this is the idiomatic way to do it in gRPC.